lnrpc/routerrpc/router_backend.go

```diff
@@ -620,6 +620,7 @@ func (r *RouterBackend) extractIntentFromSendRequest(
 		payIntent.RouteHints = append(
 			payIntent.RouteHints, payReq.RouteHints...,
 		)
+		payIntent.InvoiceFeatures = payReq.Features
```

Comment:
If not using a pay req, can mpp be specified in a separate field?

Reply:
yes, tho i was thinking of leaving that to a different pr. my main goal here is to add safety to the 99% of use cases

Reply:
I am not sure if wallets actually pass in the payment request or decode it first and pass in separate components. I don't know how much extra code it is to make it consistent? I think it should be a single `bool` field, or is there more that is required? We also have this gap possibly for signaling the tlv capability of the recipient.

Reply:
Another way to learn the flat features?

Reply:
> I think it should be a single bool field, or is there more that is required?

there are three cases for feature bits: no support, optional, or required. i think there is also a larger discussion to be had about what level of control we offer over feature bits. do we validate and sanity check for invalid or unsafe combinations? so it would be a little more than a bool at the rpc level to do it properly. 

> We also have this gap possibly for signaling the tlv capability of the recipient.

what do you mean?

Reply:
Suppose your graph is not up to date and you want to sendpayment to a node that requires tlv. 

Ideally it should be possible to signal all the same feature bits in a dedicated sendpayment field that we also have in the invoice.

Reply:
i see the use case, and agree that it both _is_ possible and _should_ be feature we support. my point tho is that i think there is some design questions as to what that interface might look like via rpc that would make it out of scope for this pr which is just meant to handle the most widely used method of sending payments, in which the negotiation is done via invoices